tests/unit/test_charm.py

```diff
@@ -269,3 +313,249 @@ def test_given_any_config_when_on_config_changed_then_systemd_manager_configurat
+            "route del default", shell=True, stdout=-1, encoding="utf-8"
+        )
+
+    @patch("subprocess.run")
+    def test_given_on_remove_default_gw_action_when_default_gw_action_then_sets_action_result(
+        self, patch_subprocess_run
+    ):
+        mock_event = Mock()
+
+        self.harness.charm._on_remove_default_gw_action(mock_event)
+
+        self.assertEqual(
+            mock_event.set_results.call_args,
+            call({"status": "ok", "message": "Default route removed!"}),
+        )
+
+    @patch("subprocess.run")
+    def test_given_on_remove_default_gw_action_when_remove_default_gw_action_then_status_does_not_change(  # noqa: E501
+        self, _
+    ):
+        mock_event = Mock()
+        old_status = self.harness.charm.unit.status
+
+        self.harness.charm._on_remove_default_gw_action(mock_event)
+
+        self.assertEqual(self.harness.charm.unit.status, old_status)
+
+    @patch("subprocess.run")
+    @patch("builtins.open", new_callable=mock_open)
+    def test_given_unit_in_relation_data_and_mme_is_not_ipv4_when_mme_relation_changed_then_status_does_not_change(  # noqa: E501
+        self, _, __
+    ):
+        old_status = self.harness.charm.unit.status
+        mock_event = Mock()
+        mock_event.unit = "lte-vepc/0"
+        mme_relation_id = self.harness.add_relation(relation_name="mme", remote_app="lte-vepc")
+        self.harness.add_relation_unit(mme_relation_id, "lte-vepc/0")
+        self.harness.update_relation_data(
+            relation_id=mme_relation_id,
+            app_or_unit="lte-vepc/0",
+            key_values={"mme-addr": "not an ip"},
+        )
+
+        self.assertEqual(self.harness.charm.unit.status, old_status)
+
+    @patch("subprocess.run", new=Mock())
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("charm.service_active")
+    def test_given_unit_in_relation_data_and_mme_is_ipv4_when_mme_relation_changed_then_status_is_active(  # noqa: E501
+        self, patch_service_active, _
+    ):
+        valid_mme = "0.0.0.0"
+        mock_event = Mock()
+        mock_event.unit = "lte-vepc/0"
+        self.harness.charm._stored.installed = True
+        self.harness.charm._stored.started = True
+        patch_service_active.return_value = True
+        self.harness.charm._stored.mme_addr = valid_mme
+
+        mme_relation_id = self.harness.add_relation(relation_name="mme", remote_app="lte-vepc")
+        self.harness.add_relation_unit(mme_relation_id, "lte-vepc/0")
+        self.harness.update_relation_data(
+            relation_id=mme_relation_id,
+            app_or_unit="lte-vepc/0",
+            key_values={"mme-addr": valid_mme},
+        )
+
+        self.assertEqual(
+            self.harness.charm.unit.status, ActiveStatus("srsenb started. mme: 0.0.0.0. ")
+        )
+
+    @patch("utils.service_restart")
+    @patch("subprocess.run")
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("charm.SrsLteCharm._get_current_status")
```

Comment:
Do we need to depend on `_get_current_status`?

Reply:
No, we don't. Actually, it was a leftover. Removed.